scripts/new-chain.ts

```diff
@@ -0,0 +1,187 @@
+			throwNewChainError(ScriptErrorCode.FACTORY_NOT_ADDED_TO_REPO, summary.ERROR as string)
+		}
+		// TODO: Create a PR to add the artifact to the repository
+	} else if (nonce > 0) {
+		summary.ERROR = getNewChainErrorMessage(ScriptErrorCode.FACTORY_DEPLOYER_ACCOUNT_NONCE_BURNED)
+		throwNewChainError(ScriptErrorCode.FACTORY_DEPLOYER_ACCOUNT_NONCE_BURNED, summary.ERROR as string)
+	} else {
+		// Get the gas price and gas limit
+		const gasPrice = await provider.getGasPrice()
+		if(!gasPrice) {
+			summary.ERROR = getNewChainErrorMessage(ScriptErrorCode.GAS_PRICE_NOT_RETRIEVED)
+			throwNewChainError(ScriptErrorCode.GAS_PRICE_NOT_RETRIEVED, summary.ERROR as string)
+		}
+		let gasLimit;
+		try {
+			gasLimit = await provider.estimateGas({
+				from: SIGNER,
+				data: FACTORY_BYTECODE,
+			})
+			if(!gasLimit) {
+				summary.ERROR = getNewChainErrorMessage(ScriptErrorCode.GAS_LIMIT_NOT_ESTIMATED)
+				throwNewChainError(ScriptErrorCode.GAS_LIMIT_NOT_ESTIMATED, summary.ERROR as string)
+			}
+		} catch (error) {
+			summary.ERROR = getNewChainErrorMessage(ScriptErrorCode.GAS_LIMIT_ESTIMATION_FAILED)
+			throwNewChainError(ScriptErrorCode.GAS_LIMIT_ESTIMATION_FAILED, summary.ERROR as string)
+		}
+
+		const gasEstimate = gasPrice.mul(gasLimit!).mul(15).div(10) // 15% buffer
+		summary.gasPrice = gasPrice.toString()
+		summary.gasLimit = gasLimit?.toString()
+		summary.gasEstimate = gasEstimate.toString()
+
+		// Get the deployed bytecode simulation
+		let simulation: string = '';
+		try {
+			simulation = await provider.call({
+				from: SIGNER,
+				data: FACTORY_BYTECODE,
+			})
+		} catch (error) {
+			summary.ERROR = getNewChainErrorMessage(ScriptErrorCode.DEPLOYMENT_SIMULATION_FAILED)
+			throwNewChainError(ScriptErrorCode.DEPLOYMENT_SIMULATION_FAILED, summary.ERROR as string)
+		}
+		summary.simulation = simulation
+		const simulationCodehash = ethers.utils.keccak256(simulation)
+		summary.simulationCodehash = simulationCodehash
+		// Check if the simulation codehash matches the expected codehash
+		if (simulationCodehash !== CODEHASH) {
+			summary.ERROR = getNewChainErrorMessage(ScriptErrorCode.FACTORY_DEPLOYMENT_SIMULATION_DIFFERENT_BYTECODE)
+			throwNewChainError(ScriptErrorCode.FACTORY_DEPLOYMENT_SIMULATION_DIFFERENT_BYTECODE, summary.ERROR as string)
+		}
+
+		// Check if the deployer account has enough balance
+		const balance = await provider.getBalance(SIGNER)
+		summary.balance = ethers.utils.formatEther(balance)
+		if (balance.lt(gasEstimate)) {
+			summary.ERROR = getNewChainErrorMessage(ScriptErrorCode.PREFUND_NEEDED, [gasEstimate.toString()])
+			throwNewChainError(ScriptErrorCode.PREFUND_NEEDED, summary.ERROR as string)
+		}
+	}
+	summary.SUCCESS = true
+	summary.RESPONSE = `**✅ Success:**<br>The issue description is valid:<br>- The RPC URL is valid<br>- The chain is in the chainlist<br>- The deployer address is pre-funded<br>:sparkles: The team will be in touch with you soon :sparkles:`
```

Comment:
if we just use `summary.reply` instead of `ERROR` and `RESPONSE`, it removes the need for the `if` in the workflow file.

Reply:
Nice idea, updated 👍🏾